lib/utils.sh

```diff
@@ -63,3 +57,27 @@ filter_version_candidates() {
   done
 }
 
+versions_cache_dir="${ASDF_DATA_DIR:-${ASDF_HOME:-$HOME/.asdf}}/tmp/$(plugin_name)/cache"
+mkdir -p $versions_cache_dir
```

Comment:
We need to quote this interpolation because ASDF_DATA_DIR, ASDF_HOME and plugin_name may contain spaces in it
```suggestion
mkdir -p "$versions_cache_dir"
```